build/sonic_build_script.sh

```diff
@@ -0,0 +1,346 @@
+#!/bin/bash
+
+# This script is used to automate the build process of Sonic for Marvell platforms
+# NOTE: Change CACHE_DIR and ARTIFACTS_DIR as per your requirement.
+
+# arguments
+BUILD_SAISERVER="N"
+BUILD_RPC="N"
+OTHER_BUILD_OPTIONS=""
+NO_CACHE="N"
+#ADMIN_PASSWORD="mrvl-sonic"
+
+DIR=$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )
+# Set MIRROR to the public mirror for the build
+MIRROR="publicmirror.azurecr.io"
+VERSION_CONTROL_COMPONENTS="deb,py2,py3,web,git,docker"
+REL_BUILD_TSTAMP=$(date +%s)
+CACHE_DIR=/var/cache/sonic-mrvl
+ARTIFACTS_DIR=/sonic-artifacts
+
+#set -e
+
+print_usage()
+{
+    echo "Usage:"
+    echo ""
+    echo " $0 -b <branch> -p <platform> -a <arch> [-c <sonic-buildimage_commit>] [--patch_script <http_path_of_patch_script>] [--url <sonic-buildimage_url>] [-s] [-r] [--mark_no_del_ws] [--no-cache] [--other_build_options <sonic_build_options>]" 
+    echo ""
+    echo "				-s : Build docker saiserver v2"
+    echo "				-r : ENABLE_SYNCD_RPC=y"
+    echo "				-c : checkout commit id"
+    echo "				--no-cache: Build without any pre cache"
+    echo "				--mark_no_del_ws: Do not cleanup ws during cleanup"
+    echo "				--other_build_options: Other sonic build options"
+}
+
+parse_arguments()
+{
+    while [[ $# -gt 0 ]]; do
+        case $1 in
+            -b|--branch)
+                BRANCH="$2"
+                shift # past argument
+                shift # past value
+                ;;
+            -p|--platform)
+                BUILD_PLATFORM="$2"
+                shift # past argument
+                shift # past value
+                ;;
+            -a|--arch)
+                BUILD_PLATFORM_ARCH="$2"
+                shift # past argument
+                shift # past value
+                ;;
+            -c|--commit)
+                BRANCH_COMMIT="$2"
+                shift # past argument
+                shift # past value
+                ;;
+            -s|--saiserver)
+                BUILD_SAISERVER="Y"
+                shift # past argument
+                ;;
+            -r|--rpc)
+                BUILD_RPC="Y"
+                shift # past argument
+                ;;
+            --no-cache)
+                NO_CACHE="Y"
+                shift # past argument
+                ;;
+            --mark_no_del_ws)
+                NO_DEL_WS="Y"
+                shift # past argument
+                ;;
+            --patch_script)
+                PATCH_SCRIPT_URL="$2"
+                shift # past argument
+                shift # past value
+                ;;
+            --url)
+                GIT_HUB_URL="$2"
```

Comment:
Please add one more argument for username and password. As current Innovium images does not use default password.

Reply:
--admin_password will be added

Reply:
looks good